src/environment.cpp

```diff
@@ -275,13 +284,19 @@ void Environment::continueRaycast(RaycastState *state, PointedThing *result)
 		state->m_previous_node = state->m_iterator.m_current_node_pos;
 		state->m_iterator.next();
 	}
-	// Return empty PointedThing if nothing left on the ray
-	if (state->m_found.empty()) {
-		result->type = POINTEDTHING_NOTHING;
-	} else {
-		*result = state->m_found.top();
+	
+	// Skip through not pointable, stop at blocking and return empty PointedThing.
+	while(!state->m_found.empty()) {
+		*result_p = state->m_found.top();
 		state->m_found.pop();
+		if (result_p->pointability == POINTABLE) {
+			return;
+		} else if (result_p->pointability == POINTABLE_BLOCKING) {
+			result_p->type = POINTEDTHING_NOTHING;
+			return;
+		}
 	}
```

Comment:
If `getSelectedActiveObjects` didn't add the not pointable objects, the loop here would not be necessary. And the `lastIndex` optimisation above would work again (so we only have to iter through the nodes till the nearest pointable object's position).

Reply:
I introduced this when I added the blocking pointable type, but you are right this can be optimized.
I removed the loop and made getSelectedActiveObjectsm return only pointable (and blocking pointable) types like it was before.